.github/workflows/java.yml

```
@@ -122,14 +120,14 @@ jobs:
                       java/client/build/reports/spotbugs/**
 
     build-amazonlinux-latest:
-        if: github.repository_owner == 'valkey-io'
+        if: (github.repository_owner == 'valkey-io' && github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' && github.event.inputs.full-matrix == 'true')
```

```suggestion
        if: github.repository_owner == 'valkey-io' && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' && github.event.inputs.full-matrix == 'true')
```
Please ensure that logical ANDs and ORs are applied in the right order.

---

Same in other workflows

---

+1